include/cantera/base/config.h.in

```diff
@@ -48,6 +48,12 @@ typedef int ftnlen;       // Fortran hidden string length type
 {CT_USE_SYSTEM_YAMLCPP!s}
 {CT_USE_DEMANGLE!s}
 
+//--------- library symbol export ---------------------------------
+
+// Export library symbols; the variable is defined for all
+// platforms in order to simplify the preprocessor setup
+{CT_DLL_EXPORT!s}
+
```

Comment:
To save you some time rediscovering how this works, the awkward structure around `CANTERA_USE_INTERNAL` is unfortunately necessary. When building the library, the compiler needs to see `__declspec(dllexport)` so it will export the functions, and when compiling code that uses the library, it needs to see `__declspec(dllimport)` to know that the symbol should be loaded from a library. There are essentially two reasonable ways of handling this:
- Use a `#define` that is given only in the Cantera `.cpp` files, as is done for the Cantera C library. This works okay for a small number of `.cpp` files.
- Use a define that is specified as a compiler command line switch, e.g. `/DCT_DLL_EXPORT` only while compilng the library components. I think this would make more sense for the much larger C++ API.

Reply:
:+1: ... thanks for the context - I was currently looking into this! (although I'm aware of the import/export mechanism). 

I put things into `config.h.in` for the moment as I believe that it isn't packaged. I was also assuming that the environment flag isn't set by `Cantera.mak` or discovered by `pkgconfig`? If that is the case, then of course I need to take a different route (like the second option you have indicated).

Reply:
Alright. While I confirmed that things work as intended, the `config.h.in` route isn't workable as it will be always included via `ct_defs.h`. I'll implement option 2 ...